test/server/cards/01_SOR/units/GideonHaskRuthlessLoyalist.spec.ts

```diff
@@ -0,0 +1,45 @@
+describe('Gideon Hask, Ruthless Loyalist', function () {
+    integration(function (contextRef) {
+        describe('Gideon Hask\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['wampa', 'battlefield-marine'],
+                    },
+                    player2: {
+                        hand: ['vanquish'],
+                        groundArena: ['gideon-hask#ruthless-loyalist', 'specforce-soldier', 'atst'],
+                    }
+                });
+            });
+
+            it('should give an Experience to a friendly unit', function () {
+                const { context } = contextRef;
+
+                // Defeating an enemy unit
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.atst);
+
+                expect(context.player2).toBeAbleToSelectExactly([context.specforceSoldier, context.gideonHask, context.atst]);
+                context.player2.clickCard(context.gideonHask);
+
+                expect(context.gideonHask).toHaveExactUpgradeNames(['experience']);
+            });
+
+            it('should give an Experience to a friendly unit when enemy unit is defeated by event', function () {
+                const { context } = contextRef;
+
+                // Defeating an enemy unit using Vanquish
```

Comment:
Thanks for the good test cases! Could we add a couple more just to be thorough (these are helpful for when we make engine changes to ensure nothing broke):

- Check that when a friendly unit dies, nothing happens (can use `expect(context.player1).toBeActivePlayer();` to confirm that the action has finished resolving with no prompt)
- Confirm that that ability still activates correctly even if Gideon dies in the combat with the enemy unit

Reply:
Addional test cases added.